fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java

```diff
@@ -570,78 +568,25 @@ private void checkPartitionColumnExprs(CreateMaterializedViewStatement statement
             }
         }
 
-        private static class ExprShuttleContext {
-            private Expr parent;
-            private int childIdx;
-
-            public ExprShuttleContext(Expr parent, int childIdx) {
-                this.parent = parent;
-                this.childIdx = childIdx;
-            }
-        }
-
+        /**
+         * Resolve the materialized view's partition expr's slot ref.
+         * @param partitionColumnExpr : the materialized view's partition expr
+         * @param connectContext      : connect context of the current session.
+         * @param queryStatement      : the sub query statment that contains the partition column slot ref
+         * @return                    : return the resolved partition expr.
+         */
         private Expr resolvePartitionExpr(Expr partitionColumnExpr,
                                           ConnectContext connectContext,
                                           QueryStatement queryStatement) {
-            if (partitionColumnExpr instanceof SlotRef) {
-                return resolvePartitionExprOfSlotRef((SlotRef) partitionColumnExpr, connectContext, queryStatement);
+            Expr expr = AnalyzerUtils.resolveExpr(partitionColumnExpr, queryStatement);
+            SlotRef slot;
+            if (expr instanceof SlotRef) {
+                slot = (SlotRef) expr;
             } else {
-                final AstVisitor exprShuttle = new AstVisitor<Void, ExprShuttleContext>() {
```

Comment:
When the partition column is a function expr, why this cannot resolve it?

Reply:
It didn't set the type correctly, but this pull request doesn't primarily fix it. Instead, I've moved the logic to AnalyzerUtils, and I'll use it in the next feature, as I mentioned in the issue.

Reply:
OK. It's better to split  refactor and bugfix into two PRs for better review if possible.

Reply:
I almost rewrote the entire function. It may not make sense to split it.